include/GafferImageUI/ImageGadget.h

```diff
@@ -339,12 +339,11 @@ class GAFFERIMAGEUI_API ImageGadget : public GafferUI::Gadget
 
 		std::vector<uint32_t> m_selectedIDs;
 
-		class BufferTexture;
-
+		class GAFFERIMAGEUI_API BufferTexture;
```

Comment:
I'm baffled by this. We have lots of similar seeming private classes elsewhere, and they don't need the export. Any ideas @ericmehl?

Reply:
That was my initial thought, and why I didn't add this initially. But I'm pretty sure that in other places where this isn't needed, we only expose raw pointers to the private class? Here, because we expose a unique_ptr to the internal class, I guess other things need to know about the destructor? Does feel a bit ugly to be exporting this, when we want it to be as private as possible.

Reply:
We use them with `unique_ptr` too - the only requirement I'm aware of there is that the main class destructor is in the `.cpp` so that the private class destructor is available when it's compiled. I don't get why it doesn't work here. Not the biggest deal, but I would like to understand it if possible - @ericmehl, any ideas?

Reply:
Turns out that not having visibility declaration in the header is fine. The problem was that the .cpp contained a visibility declaration that had been copied from GafferUI, so listed the wrong module. Fixed here: [5320f77](https://github.com/GafferHQ/gaffer/pull/6329/commits/5320f776de0d82d40b0e717a66580cf4d4a277e2)